addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -148,24 +174,56 @@ options.registry.newsletter_popup = options.registry.mailing_list_subscribe.exte
      * @override
      */
     destroy: function () {
-        this.$target.off('.newsletter_popup_option');
+        this.$target.find('.o_newsletter_content').empty();
         this._super.apply(this, arguments);
     },
 
     //--------------------------------------------------------------------------
     // Options
     //--------------------------------------------------------------------------
 
+    /**
+     * Allows to select mailing list.
+     *
+     * @see this.selectClass for parameters
+     */
+    selectMailingList(previewMode, widgetValue, params) {
+        this.$target.attr("data-list-id", widgetValue);
+        this.$target.removeData('content');
+        return this._refreshPublicWidgets();
+    },
     /**
      * @override
      */
-    select_mailing_list: function () {
-        var self = this;
-        return this._super.apply(this, arguments).then(function () {
-            self.$target.data('quick-open', true);
-            self.$target.removeData('content');
-            return self._refreshPublicWidgets();
-        });
+    onBuilt() {
+        this._super();
+        const mailingListID = parseInt(this.$target.attr('data-list-id')) || this.defaultMailingID;
+        if (mailingListID) {
```

Comment:
```suggestion
        if (this.defaultMailingID) {
```
a = b.data(x)
b.data(x, a)
Is just reapplying the same property?